Utilities/StrFmt.h

```diff
@@ -363,7 +363,10 @@ namespace fmt
 	template <typename CharT, usz N, typename... Args>
 	struct throw_exception
 	{
+		struct args_break_t {};
+
 		[[noreturn]] FORCE_INLINE SAFE_BUFFERS() throw_exception(const CharT(&fmt)[N], const Args&... args,
+			args_break_t = args_break_t{},
```

Comment:
What is this? Where is it used?

Reply:
Its not, its a barrier.

fmt::throw_exception("arg1=0x%x, arg2=0x%x (in file xxx:%u[:%u])", u32{arg1}, u32{arg2}, line, row); was actually behaving like fmt::throw_exception("arg1=, arg2= (in file xxx:%u[:%u])", line, row); Due to C++ rules of variadic arguments list. The tag type prevents that spill of arguments unless someone excplitly passes args_break_t {} at the end which is stupid.

Reply:
Its not, its a barrier.

`fmt::throw_exception("arg1=0x%x, arg2=0x%x (in file xxx:%u[:%u])", u32{arg1}, u32{arg2}, line, row);` was actually behaving like `fmt::throw_exception("arg1=, arg2= (in file xxx:%u[:%u])", /*line*/ arg1, /*row*/ arg2);` Due to C++ rules of variadic arguments list. The tag type prevents that spill of arguments unless someone excplitly passes args_break_t {} at the end which is stupid.

Reply:
just replace the whole thing with const std::source_location location = std::source_location::current().

I tried to do it last time, but I think it wasn't supported by each compiler yet.
Maybe it is now.

Reply:
It turns out std::source_location is not manually constructible from arguments which is a downgrade so as it stands it is an undesirable replacement. This feature does get used in numerous places.

Reply:
I can however do a partrial replacement if I think about it.

Reply:
Nah I just replaced the old uses and don it, the args_break_t is needed though in case std::source_location is the last argument passed to throw_exception.